### PR TITLE
Add more Knative configs into turing-init for the setting up of Turing's infrastructure

### PR DESCRIPTION
**Pull request by ashwinath**
This PR allows some extra configuration of the Knative service during the provisioning of third party software handled by the turing-init chart:
1. Allow extra Knative domains to be used
2. Allow custom container registries to be used.

This flow is tested using the e2e test flow.

### CODE REVIEW

File: infra/charts/turing-init/README.md [hunk 1]
```diff
@@ -20,8 +20,10 @@ A Helm chart for Kubernetes
...
 | istioVersion | string | `"1.9.9"` | Istio version to use |
-| knativeIstioVersion | string | `"0.18.1"` |  |
+| knativeDomains | string | `""` | Knative domains, comma seperated values, i.e. www.example.com,www.gojek.com |
+| knativeIstioVersion | string | `"0.18.1"` | Knative Istio Version to use |
 | knativeVersion | string | `"0.18.3"` | Knative Version to use |
+| knative_registries_skipping_tag_resolving | string | `""` | Knative registries skipping tag resolving, comma seperated values, i.e. www.example.com,www.gojek.com |
```
Comment (romanwozniak) [inline infra/charts/turing-init/README.md:26]: Can we have a better structure of the values file by grouping knative/istio properties under the root-level parent config groups:
```yaml
image:
  tag: ...

istio:
  version: ...
  operatorConfig: ...

knative:
  version: ...
  domains: ...
  istioVersion: ...
  registriesSkippingTagResolving: ...

spark-operator: 
  image: ...
  
```

Comment (romanwozniak) [reply infra/charts/turing-init/README.md:26]: Why is it snake here, shouldn't it be a camelCase instead?

**Review by romanwozniak — COMMENTED** [review]
Nice! Have some suggestion on the structure of config, but rather than that LGTM

File: infra/charts/turing-init/README.md (hunk 1 above)
Comment (ashwinath) [reply infra/charts/turing-init/README.md:26]: > Why is it snake here, shouldn't it be a camelCase instead?

Should have been regenerated, its supposed to be camel case, i'll regenerate this after i do the changes.

**Review by romanwozniak — APPROVED** [review]
All good, LGTM

**Pull request merged**